### ISSUE TITLE
Chinese path is not supported

### ISSUE THREAD

**Issue by imylover** [Ruslan-B/FFmpeg.AutoGen#96]
**Note: for support questions, please use [stackoverflow](https://stackoverflow.com/search?tab=newest&q=ffmpeg%20autogen)**. This repository's issues are reserved for feature requests and bug reports.

* **Windows 10 Chinese Edition:**
Not Supported Audio Devices if the path has Chinese character。
- version: 3.4

**Comment by Ruslan-B** [comment]
It could be the case as strings are marshaled as ANSI null terminated (LPStr). I guess I can change it to LPUTF8Str - assuming this will address the problem have to migrate to 4.0.

**Comment by Ruslan-B** [comment]
I can prepare private package for you for testing, as it seams there is no official support for LPUTF8Str yet in .NET (neither core https://github.com/dotnet/corefx/issues/7804  or standard https://github.com/dotnet/standard/issues/595).  
It works in a form of hack [MarshalAs((UnmanagedType)48)] instead of [MarshalAs(UnmanagedType.LPStr)].
Besides it is unclear how it is going to work in case of .net45 as I'm testing on .net471.

**Comment by Ruslan-B** [comment]
If you don't mind I'll rename the issue into interop unicode support? As it concerns not only Chinese.

**Comment by imylover** [comment]
With your help,I solved the problem. I'm not sure this problem will happen in other languages. I think the problem arises because of Unicode.  Thank you again.

**Comment by Ruslan-B** [comment]
@KFAFSP thank for the hint I'll try to do so.

**Comment by beefsteak** [comment]
I have the same problem，How to solve it？

**Comment by Ruslan-B** [comment]
Update to most recent package please.

**Comment by catester** [comment]
For reference, this change has a side effect (for any file name not related to chinese) and it causes the below error when run on .net40 runtime (custom build target, did not test .net45 runtime though):

```
System.Runtime.InteropServices.MarshalDirectiveException: Cannot marshal 'parameter #2': Invalid managed/unmanaged type combination (String parameters and return types must be paired with LPStr, LPWStr, LPTStr, BStr, TBStr, VBByRefStr, or AnsiBStr).
   at System.Runtime.InteropServices.Marshal.GetDelegateForFunctionPointerInternal(IntPtr ptr, Type t)
   at System.Runtime.InteropServices.Marshal.GetDelegateForFunctionPointer(IntPtr ptr, Type t)
   at FFmpeg.AutoGen.Native.FunctionLoader.GetFunctionDelegate[T](IntPtr nativeLibraryHandle, String functionName, Boolean throwOnError)
   at FFmpeg.AutoGen.ffmpeg.avformat_open_input(AVFormatContext** ps, String url, AVInputFormat* fmt, AVDictionary** options)
```

Resolved it by reverting to version 4.1.0.3.
I hope there is a more compatible way to do this change. I know you don't even target net40 but just in case.

> It works in a form of hack [MarshalAs((UnmanagedType)48)] instead of [MarshalAs(UnmanagedType.LPStr)].
> Besides it is unclear how it is going to work in case of .net45 as I'm testing on .net471.

**Comment by Ruslan-B** [comment]
Right it is runtime version specific i.e. version of .net installed. I can add support of net40 obviously
but then it will not support any unicode at all. Might be the good time to introduce conditional compiling.
